src/Color/Convert.elm

```diff
@@ -133,22 +132,53 @@ Converts a string to `Maybe` of color.
 -}
 hexToColor : String -> Result String Color
 hexToColor =
-    String.toLower
-        >> Regex.find (Regex.AtMost 1)
-            (Regex.regex "^#?([a-f\\d]{2})([a-f\\d]{2})([a-f\\d]{2})$")
-        >> List.map .submatches
-        >> List.head
-        >> Result.fromMaybe "Parsing hex regex failed"
-        >> Result.andThen
-            (\rs ->
-                case List.map (Maybe.map parseIntHex) rs of
-                    (Just (Ok r)) :: (Just (Ok g)) :: (Just (Ok b)) :: [] ->
-                        Ok <| rgb r g b
-
-                    _ ->
-                        -- there could be more descriptive error cases per channel
-                        Err "Parsing ints from hex failed"
-            )
+    let
+        {- Converts "f" to "ff" and "ff" to "ff" -}
+        extend : String -> String
+        extend token =
+            case String.toList token of
+                [ token ] ->
+                    String.fromList [ token, token ]
+
+                _ ->
+                    token
+
+        {- Converts "<pattern>" to "(<pattern><pattern><pattern>)" -}
+        buildColorPattern : String -> String
```

Comment:
What about this?
![image](https://cloud.githubusercontent.com/assets/7101648/22906564/11f57bb0-f256-11e6-9f93-a041189b4471.png)